pymongo/bulk.py

```diff
@@ -317,10 +329,10 @@ def _execute_command(self, generator, write_concern, session,
                         # Synthesize the full bulk result without modifying the
                         # current one because this write operation may be retried.
                         full = copy.deepcopy(full_result)
-                        _merge_command(run, full, run.idx_offset, result)
+                        _merge_command(run, full, run.idx_offset, result, codec_options)
```

Comment:
One problem with this approach is that it doesn't handle `_id` that are subdocuments, eg: `_id={'f': Binary(), 'f2': ...}`. Could you add a test for this as well?

Instead of converting the result after the fact, we should ensure that BulkWriteContext.execute() decodes the command response with the correct uuid_representation/CodecOptions to begin with. This will involve passing the Collection's `__write_response_codec_options` to the BulkWriteContext to later use for decoding.

Reply:
Cool, fixed the decoding in the latest commit.  I'll address the tests in the morning.

Reply:
Added the subdocument test